### PR TITLE
pytest migration of unit.common.metadata.test_CoordMetadata

### PR DESCRIPTION
**Pull request by bjlittle**
### Description

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

---
Add any of the below labels to trigger actions on this PR:

- https://github.com/SciTools/iris/labels/benchmark_this

### CODE REVIEW

**Review by pp-mo — CHANGES_REQUESTED** [review]
Couple of glitches, 
plus some almost-automatic improvements that occur in a multiple places.

Perhaps I should fix those myself, combined for your approval in a PR-to-PR ?

**Comment by bjlittle** [comment]
@pp-mo Thanks. Back to you 👍

File: lib/iris/tests/unit/common/metadata/test_CoordMetadata.py [hunk 1]
```diff
@@ -218,125 +215,126 @@ def setUp(self):
...
             right[member] = None
             rmetadata = self.cls(**right)
             expected = right.copy()
 
-            with mock.patch("iris.common.metadata._LENIENT", return_value=True):
-                self.assertTrue(expected, lmetadata.combine(rmetadata)._asdict())
-                self.assertTrue(expected, rmetadata.combine(lmetadata)._asdict())
+            mocker.patch("iris.common.metadata._LENIENT", return_value=True)
+            assert expected, lmetadata.combine(rmetadata)._asdict()
+            assert expected, rmetadata.combine(lmetadata)._asdict()
```
[the remark this answers is not in the document]
Comment (pp-mo) [reply lib/iris/tests/unit/common/metadata/test_CoordMetadata.py:323]: This one still not fixed ?

**Comment by pp-mo** [comment]
I think you missed a tiny one -- see above.
Otherwise, ready to go now!

**Comment by pp-mo** [comment]
Thanks @bjlittle for staying the distance on this one !

**Pull request merged**